contracts/Donations.sol

```diff
@@ -0,0 +1,193 @@
+        address _to
+    ) external onlyRole(WORKER_ROLE) {
+        require(
+            transferableAmounts[_token][_destinationId] != 0,
+            "Donations: nothing to donate"
+        );
+
+        uint256 amount = transferableAmounts[_token][_destinationId];
+        transferableAmounts[_token][_destinationId] = 0;
+
+        _token.safeTransfer(_to, amount);
+
+        emit DonationsSent(_destinationId, _token, _to, amount);
+    }
+
+    /**
+     * Mints an NFT for every donation in @param _params.
+     * The @param _groupId is used to uniquely identify this group of donations.
+     * Ideally, @param _groupId is the hash of the transaction where the yield for the donations was claimed to by the treasury.
+     *
+     * @param _groupId Unique identifier for the group of donations in @param _params.
+     * @param _params Donation params.
+     */
+    function mint(bytes32 _groupId, DonationParams[] calldata _params)
+        external
+        onlyRole(WORKER_ROLE)
+    {
+        require(
+            !processedDonationsGroups[_groupId],
+            "Donations: already processed"
+        );
+
+        uint64 expiry = _getBlockTimestamp() + ttl;
+        uint256 length = _params.length;
+
+        for (uint256 i = 0; i < length; i++) {
+            uint256 _metadataId = metadataId.current();
+            metadataId.increment();
+
+            metadata[_metadataId] = Metadata({
+                destinationId: _params[i].destinationId,
+                token: _params[i].token,
+                expiry: expiry,
+                amount: _params[i].amount
+            });
+
+            _mint(_params[i].owner, _metadataId);
+
+            emit DonationMinted(
+                _metadataId,
+                _params[i].destinationId,
+                _groupId,
+                _params[i].token,
+                expiry,
+                _params[i].amount,
+                _params[i].owner
+            );
+        }
+
+        processedDonationsGroups[_groupId] = true;
+    }
+
+    /**
+     * Burns the NFT and sets the amount donated to be transferred to the charity.
+     *
+     * @param _id ID of the NFT.
+     */
+    function burn(uint256 _id) external {
+        require(
+            ownerOf(_id) == _msgSender() || hasRole(WORKER_ROLE, _msgSender()),
+            "Donations: not allowed"
+        );
+
+        uint256 destinationId = metadata[_id].destinationId;
+        IERC20 token = metadata[_id].token;
+        uint256 amount = metadata[_id].amount;
+
+        transferableAmounts[token][destinationId] += amount;
+
+        _burn(_id);
+
+        emit DonationBurned(_id);
+    }
+
+    function _getBlockTimestamp() private view returns (uint64) {
+        return uint64(block.timestamp);
+    }
+
+    function supportsInterface(bytes4 interfaceId)
+        public
+        view
+        virtual
+        override(AccessControl, ERC721)
+        returns (bool)
+    {
+        return super.supportsInterface(interfaceId);
+    }
```

Comment:
if you're just calling super, why override this in the first place?

Reply:
maybe you actually want `return interfaceId = type(ERC721).interfaceId || super.supportsInterface(interfaceId)`

Reply:
Honestly, I have no idea. I copied this from one of the ERC721 contracts that have more stuff inside on openzepplin.  I'm not sure what I want so I trust you know better